Vlad_B/app.py

```diff
@@ -0,0 +1,122 @@
+    ('ME', [': ', '55 ', ': ', ': ', ': ', ': ', '71 ', '72 ', '74 ']),
+    ('MK', [': ', '58 ', '65 ', '68 ', '69 ', '75 ', '74 ', '79 ', '82 ']),
+    ('MT', ['75 ', '77 ', '78 ', '80 ', '81 ', '81 ', '85 ', '84 ', '86 ']),
+    ('NL', ['94 ', '94 ', '95 ', '96 ', '96 ', '97 ', '98 ', '98 ', '98 ']),
+    ('NO', ['92 ', '93 ', '94 ', '93 ', '97 ', '97 ', '97 ', '96 ', '98 ']),
+    ('PL', ['67 ', '70 ', '72 ', '75 ', '76 ', '80 ', '82 ', '84 ', '87 ']),
+    ('PT', ['58 ', '61 ', '62 ', '65 ', '70 ', '74 ', '77 ', '79 ', '81 ']),
+    ('RO', ['47 ', '54 ', '58 ', '61 ', '68 ', '72 ', '76 ', '81 ', '84 ']),
+    ('RS', [': ', ': ', ': ', ': ', '64 ', ': ', '68 ', '73 ', '80 ']),
+    ('SE', ['91 ', '92 ', '93 ', '90 ', '91 ', '94 ', '95 ', '93 ', '96 ']),
+    ('SI', ['73 ', '74 ', '76 ', '77 ', '78 ', '78 ', '82 ', '87 ', '89 ']),
+    ('SK', ['71 ', '75 ', '78 ', '78 ', '79 ', '81 ', '81 ', '81 ', '82 ']),
+    ('TR', [': ', '47 ', '49 ', '60 ', '70 ', '76 ', '81 ', '84 ', '88 ']),
+    ('UK', ['83 ', '87 ', '88 ', '90 ', '91 ', '93 ', '94 ', '95 ', '96 ']),
+    ('XK', [': ', ': ', ': ', ': ', ': ', ': ', '89 ', '93 ', '93 ']),
+]
+
+
+# function that prepares the dataset
+def data_set(country_codes, descr, data):
+    # generate a new raw data with year/coverage association
+    new_raw_data = {raw_data_tuple[0]: list(zip(descr[1], raw_data_tuple[1])) for raw_data_tuple in data}
+
+    # based on new data, generate final dataset
+    data = {}
+    for new_raw_data_key, new_raw_data_vals in new_raw_data.items():
+        data.update(
+            {new_raw_data_key: [{'year': year.strip(), 'coverage': coverage} for year, coverage in new_raw_data_vals]})
+
+    # replace specified country codes with country names and convert 'coverages' in integer
+    new_data = {}
+    for data_key, data_value in data.items():
+        for coverage_dict in data_value:
+            if coverage_dict['coverage'].find(':') != -1:
+                coverage_dict['coverage'] = None
+            else:
+                coverage_dict['coverage'] = int(coverage_dict['coverage'])
+        for code_dict in country_codes:
+            if data_key in code_dict.values():
+                new_data.update({code_dict['Name']: data_value})
+    return new_data
+
+
+# function to retrieve data for each year
+def get_year_data(dataset, year):
+    country_list = []
+    for key, val in dataset.items():
+        for coverage_dict in val:
+            if year == coverage_dict['year']:
+                country_list.append(tuple([key, coverage_dict['coverage']]))
+    return {year: country_list}
+
+
+# function to retrieve data for each country
+def get_country_data(dataset, country):
+    year_coverage = []
+    for coverage_dict in dataset[country]:
+        year_coverage.append(tuple([coverage_dict['year'], coverage_dict['coverage']]))
+    return {country: year_coverage}
+
+
+def perform_average(dataset):
+    coverage_list = [coverage for x, coverage in dataset if coverage]
```

Comment:
`x` can be replaced with `_` since we are not using it

Reply:
ok